hedera-did-method-specification.md

```diff
@@ -0,0 +1,385 @@
+
+### 3.1.4. Revoke 
+
+A property or a DID document is updated by sending a `ConsensusSubmitMessage` transaction to a Hedera network node. It is executed by sending a `submitMessage` RPC call to HCS with the `ConsensusSubmitMessageTransactionBody` containing:
+
+- `topicID` - equal to the `did-topic-id` element of the DID namestring.
+- `message` - a JSON DID message envelope described above
+  - `operation` set to `revoke`
+  - `event` payload either a `Service`, `VerificationMethod` or `VerificationRelationship` object with `id` set to the property to remove.
+
+### 3.1.5. Delete 
+
+A Whole DID document is deleted/nullified by sending `ConsensusSubmitMessage` transaction to a Hedera network node. It is executed by sending a `submitMessage` RPC call to HCS with the `ConsensusSubmitMessageTransactionBody` containing:
+
+- `topicID` - equal to the `did-topic-id` element of the DID namestring.
+- `message` - a JSON DID message envelope described above
+  - `operation` set to `revoke`
+  - `event` payload will be ignored
+
+## 3.2. Event Payload
+
+A Base64-encoded JSON object that conforms the different properties of a DID documents to the [DID Specification](https://w3c.github.io/did-core/), the accepted events are listed within this section.
+
+### 3.2.1. DID Document
+
+A Hedera DID MAY be created by creating a reference to a DID document available in [IPFS](https://ipfs.io/).
+
+`DIDDocument` event value must have a JSON structure defined by a [DIDDocument-schema](DIDDocument.schema.json) and contains the following properties:
+- `DIDDocument` - The DIDOwner event with the following attributes:
+  - `id` - The DID id
+  - `type` - The document type, MAY include the DID document serialisation representation.
+  - `cid` - The Content Identifiers to point to DID document in IPFS.
+  - `url` - A URL to the IPFS document MAY be included for convenience.
+
+```json
+{
+  "DIDDocument": {
+    "id": "did:hedera:testnet:z6MknSnvSESWvijDEysG1wHGnaiZSLSkQEXMECWvXWnd1uaJ_0.0.1723780",
+    "type": "DIDDocument",
+    "cid": "bafybeifn6wwfs355md56nhwaklgr2uvuoknnjobh2d2suzsdv6zpoxajfa/did-document.json",
+    "url": "https://ipfs.io/ipfs/bafybeifn6wwfs355md56nhwaklgr2uvuoknnjobh2d2suzsdv6zpoxajfa/did-document.json"
+  }
+}
+```
+
+### 3.2.2. DID Owner
+
+Each identifier always has a controller address. By default, it is the same as the identifier address, however the resolver must validate this against the DID document. This controller address must be represented in the DID document as a verificationMethod entry with the id set as the DID being resolved and with the fragment `#did-root-key` appended to it. A reference to it must also be added to the authentication and assertionMethod arrays of the DID document.
+
+If the controller for a particular DID is changed, a `DIDOwner` event is emitted through an DID update message. The event data MUST be used to update the `#did-root-key` entry in the verificationMethod array.
```

Comment:
If the the did-root-key is updated, then the did namestring no longer maps to the base56 encoding of the did-root-key. Is this an issue?

Perhaps define the did namestring by the "initial did-root-key"?

Reply:
Is this something we need to allow to comply with the W3C specs? 
How other specs deal with this issue? 
If I need to check the DID Document in any case, is it reasonable to use a namestring that does not contain the initial did-root-key at all like the account ID instead of the base58-pub-key?